cytominer_database/ingest_incl_parquet.py

```diff
@@ -0,0 +1,203 @@
+# ----> write a new validate_csv_set() and deal with weird image/compartment output structure         
+"""
+A mechanism to ingest CSV files into a database.
+
+In morphological profiling experiments, a CellProfiler pipeline is often run in parallel across multiple images and
+produces a set of CSV files. For example, imaging a 384-well plate, with 9 sites per well, produces 384 * 9 images;
+a CellProfiler process may be run on each image, resulting in a 384*9 output directories (each directory typically
+contains one CSV file per compartment (e.g. Cells.csv, Cytoplasm.csv, Nuclei.csv) and one CSV file for per-image
+measurements (e.g. Image.csv).
+
+``cytominer_database.ingest.seed`` can be used to read all these CSV files into a database backend. SQLite is the
+recommended engine, but ingest will likely also work with PostgreSQL and MySQL.
+
+``cytominer_database.ingest.seed`` assumes a directory structure like shown below:
+
+| plate_a/
+|   set_1/
+|       file_1.csv
+|       file_2.csv
+|       ...
+|       file_n.csv
+|   set_2/
+|       file_1.csv
+|       file_2.csv
+|       ...
+|       file_n.csv
+|   ...
+|   set_m/
+|       file_1.csv
+|       file_2.csv
+|       ...
+|       file_n.csv
+
+Example::
+
+    import cytominer_database.ingest
+
+    cytominer_database.ingest.seed(source, target, config_path)
+
+    where
+    source = path/to/plate_a
+    target = path/to/output/directory
+    config_path = path/to/plate_a/ingest_config.ini
+Notes: 
+* seed() will run regardless of where the config-file was stored, as long as its' full path is passed. 
+This is not the case for the command-line command, where the config-file is specified only by the basename 
+(e.g. ingest_config.ini) and is assumed to reside in path/to/plate_a.
+
+* 'file_1.csv', ..., 'file_n.csv' is typically equivalent to 'Cells.csv', 'Cytoplasm.csv', 'Nuclei.csv', 'Image.csv', 'Object.csv'.
+
+"""
+
+"""
+* Extended explanations of how the PARQUET writer is used (see also the readme.rst).
+[Big Picure]
+When opening a Parquet file with the Pyarrow.parquet.writer, we use a reference table to set
+one fixed table schema (pyarrow.Table.schema) (per table kind).
+[Schema compatibility: column names]
+For each new table to be written to the open Parquet writer, schema compatibility is assured by aligning the dataframes. 
+The alignment assures that the new table will hold all columns present in the reference table, however, columns not present in the 
+reference table will be dropped from the new table. 
+Hence it is important to assure that the reference table has all columns that will be present in any of the tables of its kind. 
+
+[Schema compatibility: column types]
+Furthermore, the column types of columns with identical column names must agree. This has proven to be problematic
+when software further up the processing pipeline casts the types of its' output automatically. 
```

Comment:
```suggestion
when software further up the processing pipeline casts the types of its output automatically. 
```

Reply:
Was removed - thank you for the correction though!